### ISSUE TITLE
[dynamo] Data-dependent branching (121 models, 764 breaks)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#54]
## Summary

121 models produce 764 graph breaks from this pattern.

## Break Reason

```
Graph Break Reason: Encountered graph break when attempting to trace CALL_FUNCTION_EX: a variadic function call, e.g. f(*args):
```

## Affected Models

| Model | Modes | Breaks |
|-------|-------|--------|
| AudioFlamingo3ForConditionalGeneration | train | 7 |
| AutoformerModel | train | 5 |
| BambaForCausalLM | eval, train | 12 |
| BambaModel | eval, train | 8 |
| BartForCausalLM | train | 5 |
| BartForConditionalGeneration | train | 9 |
| BartModel | train | 7 |
| BigBirdPegasusForCausalLM | train | 5 |
| BigBirdPegasusForConditionalGeneration | train | 3 |
| BigBirdPegasusModel | train | 3 |
| BioGptForCausalLM | train | 5 |
| BioGptModel | train | 3 |
| BlenderbotForCausalLM | train | 5 |
| BlenderbotForConditionalGeneration | train | 9 |
| BlenderbotModel | train | 7 |
| BlenderbotSmallForCausalLM | train | 5 |
| BlenderbotSmallForConditionalGeneration | train | 9 |
| BlenderbotSmallModel | train | 7 |
| Blip2ForConditionalGeneration | train | 7 |
| Blip2Model | train | 7 |
| CohereAsrForConditionalGeneration | train | 8 |
| CohereAsrModel | train | 6 |
| DFineModel | train | 7 |
| Data2VecAudioModel | train | 1 |
| EncodecModel | eval, train | 42 |
| FSMTForConditionalGeneration | train | 5 |
| FSMTModel | train | 4 |
| FalconH1ForCausalLM | eval, train | 12 |
| FalconH1Model | eval, train | 8 |
| FastSpeech2ConformerModel | eval, train | 4 |
| FlaubertModel | train | 1 |
| FlaubertWithLMHeadModel | train | 1 |
| Florence2ForConditionalGeneration | train | 6 |
| Florence2Model | train | 6 |
| GraniteMoeHybridForCausalLM | eval, train | 12 |
| GraniteMoeHybridModel | eval, train | 8 |
| GroundingDinoModel | train | 1 |
| HubertModel | train | 1 |
| IBertModel | train | 28 |
| InformerModel | train | 6 |
| InstructBlipForConditionalGeneration | train | 7 |
| InstructBlipVideoForConditionalGeneration | train | 7 |
| JambaForCausalLM | eval, train | 12 |
| JambaModel | eval, train | 8 |
| Kosmos2ForConditionalGeneration | train | 7 |
| Kosmos2Model | train | 7 |
| LEDForConditionalGeneration | train | 3 |
| LEDModel | train | 3 |
| M2M100ForConditionalGeneration | train | 7 |
| M2M100Model | train | 6 |
| MBartForCausalLM | train | 5 |
| MBartForConditionalGeneration | train | 4 |
| MBartModel | train | 4 |
| MMGroundingDinoModel | train | 1 |
| MarianForCausalLM | train | 5 |
| MarianMTModel | train | 9 |
| MarianModel | train | 7 |
| MiniCPM-V-Resampler | eval, train | 6 |
| MvpForCausalLM | train | 2 |
| MvpForConditionalGeneration | train | 5 |
| MvpModel | train | 4 |
| NemotronHForCausalLM | eval, train | 12 |
| NemotronHModel | eval, train | 8 |
| NllbMoeForConditionalGeneration | train | 8 |
| NllbMoeModel | train | 6 |
| OPTForCausalLM | train | 5 |
| OPTModel | train | 5 |
| OlmoHybridForCausalLM | eval, train | 12 |
| OlmoHybridModel | eval, train | 8 |
| OpenAIPrivacyFilterModel | eval, train | 2 |
| OpenVoice-SynthesizerTrn-infer | eval | 3 |
| Ovis2ForConditionalGeneration | eval, train | 2 |
| PLBartForCausalLM | train | 5 |
| PLBartForConditionalGeneration | train | 7 |
| PLBartModel | train | 6 |
| PPDocLayoutV2Model | train | 7 |
| PPDocLayoutV3Model | train | 7 |
| PegasusForCausalLM | train | 5 |
| PegasusForConditionalGeneration | train | 9 |
| PegasusModel | train | 7 |
| PegasusXForConditionalGeneration | train | 9 |
| PegasusXModel | train | 7 |
| Qwen2AudioForConditionalGeneration | train | 5 |
| Qwen3NextForCausalLM | eval, train | 12 |
| Qwen3NextModel | eval, train | 8 |
| Qwen3_5ForCausalLM | eval, train | 12 |
| Qwen3_5ForConditionalGeneration | eval, train | 6 |
| Qwen3_5Model | eval, train | 12 |
| Qwen3_5MoeForCausalLM | eval, train | 12 |
| Qwen3_5MoeForConditionalGeneration | eval, train | 6 |
| Qwen3_5MoeModel | eval, train | 12 |
| Qwen3_5MoeTextModel | eval, train | 8 |
| Qwen3_5TextModel | eval, train | 8 |
| RTDetrModel | train | 7 |
| RTDetrV2Model | train | 7 |
| RagModel | train | 10 |
| ReformerModel | eval, train | 6 |
| SEWModel | train | 1 |
| SeamlessM4TModel | train | 4 |
| SeamlessM4TTextToUnitForConditionalGeneration | train | 5 |
| SeamlessM4TTextToUnitModel | train | 4 |
| SeamlessM4Tv2Model | train | 4 |
| Speech2TextForConditionalGeneration | train | 7 |
| Speech2TextModel | train | 6 |
| SpeechEncoderDecoderModel | train | 1 |
| SpeechT5Model | train | 2 |
| TimeSeriesTransformerModel | train | 6 |
| TrOCRForCausalLM | train | 2 |
| UniSpeechModel | train | 1 |
| UniSpeechSatModel | train | 1 |
| ViltModel | eval, train | 2 |
| Wav2Vec2BertModel | train | 1 |
| Wav2Vec2ConformerModel | train | 1 |
| Wav2Vec2Model | train | 1 |
| WavLMModel | train | 1 |
| WhisperForCausalLM | train | 5 |
| WhisperForConditionalGeneration | train | 9 |
| WhisperModel | train | 7 |
| XGLMForCausalLM | train | 4 |
| XGLMModel | train | 3 |
| XcodecModel | train | 1 |

## Impact

Fixing this pattern moves **9 models** to fullgraph (models where this is the only break reason).

## Related Issues

- #2
- #8
- #10
- #14
- #16
- #18
- #19
- #26
- #27
- #28
- #55
- #56

## Tested On

- PyTorch 2.13.0.dev20260425+cu126
- Transformers 5.6.2
- Sweep date: 2026-04-26T08:48:25

**Comment by penguinwu** [comment]
## Sub-bucket breakdown (sweep 2026-04-28, animesh-fullgraph)

Re-classified the 751 occurrences observed in the 2026-04-28 sweep (PR #181552 baseline) by inspecting the data-dep computation chain in each `Graph break` reason text:

| Sub-bucket | Models | Breaks | File-line sites | Pattern | Sub-issue |
|---|---|---|---|---|---|
| Layerdrop (train) | 89 | 466 | 63 | `if dropout_probability < self.layerdrop` where `dropout_probability = torch.rand([])` or `np.random.uniform(0,1)` | #77 |
| Mamba/linear-attn mask fast-path | 20 | 204 | 9 | `attention_mask is not None and torch.all(attention_mask == 1)` in `_update_mamba_mask` / `_update_linear_attn_mask` | #78 |
| Encodec conv1d padding math | 1 | 38 | 1 | `n_frames = (length - kernel + padding) / stride + 1` then `if extra_pad > 0` (`_get_extra_padding_for_conv1d`) | (model-specific, not filed) |
| iBERT quant activation range | 1 | 27 | 2 | `if self.x_min.min() > -1.1e-5 and self.x_max.max() < 1.1e-5:` | (model-specific, not filed) |
| Reformer chunk-length padding | 1 | 6 | 4 | `if must_pad_to_match_chunk_length:` (np.lcm + seqlen) | (model-specific, not filed) |
| Other one-offs | 4 | 10 | 4 | FastSpeech2Conformer:116, OpenAIPrivacyFilter:303, Ovis2:583, Vilt:153 (no detailed src in reason text) | (model-specific) |

Total: 751 breaks across 116 models.

Sub-issues #77 and #78 cover the two cross-family patterns (layerdrop, mamba mask) that the dynamo team can address as single fixes with high leverage. The remaining sub-buckets are model-specific HF idioms — better tracked in HF transformers issues if pursued.

**Comment by penguinwu** [comment]
## Related sub-issues filed (2026-04-29)

The original #54 umbrella has now been broken up into specific sub-issues so each is closeable on its own. Sister issues from the same data-dep branching cluster:

- #77 — layerdrop pattern (89 models, 466 train breaks)
- #78 — mamba/linear-attn mask all-ones fast-path (20 models, 204 breaks)
- #85 — Encodec conv1d padding (1 model, 38 breaks)
- #86 — iBert quant activation range (1 model, 27 breaks)
- #87 — Reformer chunk-length padding (1 model, 6 breaks)

#54 is closing — its content is fully covered by the 5 sub-issues above.
